garak/generators/huggingface.py

```diff
@@ -178,10 +184,17 @@ class ConversationalPipeline(Generator, HFCompatible):
     generator_family_name = "Hugging Face 🤗 pipeline for conversations"
     supports_multiple_generations = True
 
-    def __init__(self, name, do_sample=True, generations=10, device=0):
+    def __init__(
+        self, name="", do_sample=True, generations=10, device=0, config_root=_config
+    ):
+        self.do_sample = do_sample
+        self.generations = generations
+        self.device = device
         self.fullname, self.name = name, name.split("/")[-1]
 
-        super().__init__(name, generations=generations)
+        super().__init__(
+            self.name, generations=self.generations, config_root=config_root
+        )
```

Comment:
The parent does it now that you point this out, I see there is an existing failure to pass the `do_sample` and `device` to the `Super()__initi__` here. This module seemed like it a candidate for refactor this class in particular since the call here to `Super()` will instantiate the transformer in `self.generator` and then this constructor replaces it with the `optimum` version.

I toyed with refactoring this now but felt it may deserve a distinct PR to focus acceptance testing of changes for this module.